sdk/src/main/java/io/opentelemetry/sdk/metrics/AggregatorMap.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.LabelSet;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.aggregator.Aggregator;
+import io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+abstract class AggregatorMap {
```

Comment:
This class name is confusing. Why `Map` ?

Reply:
Maybe "AggregationStrategy" would be better?

Reply:
Also, I think this would be better as an interface, since it's internal and we don't need to worry about breaking things by adding new methods.

Reply:
I am very annoyed with interfaces since I cannot have some implementations in the same file and require an extra file like Aggregator/Aggregators :(.

Reply:
yeah, java 7 is a problem. this is all fixed in java 8. Anyway, an abstract class is fine for now.